applications/jonny5/src/j5_channels.erl

```diff
@@ -328,6 +328,27 @@ per_minute_cost(AccountId) ->
                         call_cost(Channel) + Cost
                 end, 0, ets:select(?TAB, MatchSpec)).
 
+-spec real_per_minute_cost(ne_binary()) -> non_neg_integer().
+real_per_minute_cost(AccountId) ->
+    MatchSpec = [{#channel{account_id = AccountId
+                           ,account_billing = <<"per_minute">>
+                           ,_='_'
+                          }
+                  ,[]
+                  ,['$_']
+                 }
+                 ,{#channel{reseller_id = AccountId
+                            ,reseller_billing = <<"per_minute">>
+                            ,_='_'
+                           }
+                   ,[]
+                   ,['$_']
+                  }
+                ],
+    lists:foldl(fun(Channel, Cost) ->
```

Comment:
You are correct. From what I gather, doing the explicit matching in the MatchHead portion (`#channel{account_id=AccountId}`) is a faster operation than if you did the matching the guard (`#channel{account_id='$1'}, [{'==', '$1', AccountId}]`), and avoids a full table scan (as you rightly pointed out that `ets:foldl/3` would do).

I wonder, though, if we could just match `answered_timestamp` in the match spec and iterate over that in the call_cost function, instead of copying the whole channel object over from ETS to process the one field?